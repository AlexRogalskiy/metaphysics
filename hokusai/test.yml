---
version: '2'
services:
  metaphysics:
    command: yarn test
    extends:
      file: common.yml
      service: metaphysics
    environment:
    - NODE_ENV=test
    env_file: ../.env.test
