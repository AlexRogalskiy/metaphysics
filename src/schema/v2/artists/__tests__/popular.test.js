/* eslint-disable promise/always-return */
import gql from "lib/gql"
import { runQuery } from "schema/v2/test/utils"

describe("Popular Artists", () => {
  it("makes a call for popular artists", async () => {
    const query = gql`
      {
        popularArtists {
          artists {
            slug
          }
        }
      }
    `

    const context = {
      popularArtistsLoader: () =>
        Promise.resolve([
          { birthday: "1900", artworks_count: 100, id: "ortina" },
          { birthday: "1900", artworks_count: 100, id: "xtina" },
        ]),
    }

    await runQuery(query, context).then(data => {
      expect(data).toMatchSnapshot()
    })
  })
})
