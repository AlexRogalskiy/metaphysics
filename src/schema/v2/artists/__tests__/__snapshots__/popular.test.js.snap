// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Popular Artists makes a call for popular artists 1`] = `
Object {
  "popularArtists": Object {
    "artists": Array [
      Object {
        "slug": "ortina",
      },
      Object {
        "slug": "xtina",
      },
    ],
  },
}
`;
